### PR TITLE
Support setting shared memory size.

### LINKED ISSUE #1030: How to increase the share memory size for a docker image
### Cookbook version
v4.6.2 

### Chef-client version
12.5.1

### Platform Details
Centos 6.5

### Scenario:

I am using this cookbook to deploy the selenium test grid and it requires the shared memory of the image to be at a minimum of 2g (see https://github.com/SeleniumHQ/docker-selenium#running-the-images).

I wish to change the shared memory size of the image. Basically add "--shm-size=2g" to the docker run.

### Steps to Reproduce:
None. It is a general question and not a bug.

### Expected Result:
na

### Actual Result:
na

### PR DESCRIPTION
**Pull request by don-code**
### Description
This change enables setting the shared memory size of a container, as one would with `docker run --shm-size ...`.

This functionality was added in Docker 1.10.0 - this would break older Dockers. I do not know what level of compatibility this project attempts to maintain with Docker itself.

### Issues Resolved

#1030 

### Check List

- [ ] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [ ] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

**Comment by don-code** [comment]
The tests which failed are also failing against master. Is there anything else I can do to help drive this home? My urgency is a customer-facing bug that this change fixes, and I'd prefer not to run off a fork.

File: test/cookbooks/docker_test/recipes/container.rb [hunk 1]
```diff
@@ -1213,6 +1213,7 @@
   memory_swap '5M'
   memory_swappiness 50
   memory_reservation '5m'
+  shm_size '32m'
```
Comment (chasebolt) [inline test/cookbooks/docker_test/recipes/container.rb]: can you complete this test by adding the matching inspec line in `test/integration/resources/inspec/assert_functioning_spec.rb`, bottom of the file is the memory container.

**Comment by chasebolt** [comment]
thanks ill get this merged in and released!

**Comment by chasebolt** [comment]
released in `v4.9.2`

**Pull request merged**